app_dart/lib/src/service/scheduler.dart

```diff
@@ -56,6 +65,45 @@ class Scheduler {
     }
   }
 
+  /// Schedule tasks against [PullRequest].
+  ///
+  /// If [PullRequest] was merged, schedule prod tasks against it.
+  /// Otherwise if it is presubmit, schedule try tasks against it.
+  Future<void> addPullRequest(PullRequest pr) async {
+    // TODO(chillers): Support triggering on presubmit. https://github.com/flutter/flutter/issues/77858
+    if (!pr.merged) {
+      log.warning('Only pull requests that were closed and merged should have tasks scheduled');
+      return;
+    }
+
+    final String fullRepo = pr.base.repo.fullName;
+    final String repo = pr.base.repo.name;
+    final String branch = pr.base.ref;
+    final String sha = pr.mergeCommitSha;
+
+    final String id = '$fullRepo/$branch/$sha';
+    final Key<String> key = datastore.db.emptyKey.append<String>(Commit, id: id);
+    final Commit mergedCommit = Commit(
+      author: pr.user.login,
+      authorAvatarUrl: pr.user.avatarUrl,
+      branch: branch,
+      key: key,
+      // The field has a max length of 1500 so ensure the commit message is not longer.
+      message: truncate(pr.title, 1490, omission: '...'),
+      repository: repo,
+      sha: sha,
+      timestamp: pr.mergedAt.millisecondsSinceEpoch,
+    );
+
+    if (await _commitExistsInDatastore(mergedCommit)) {
```

Comment:
Yes. Since commits can be scheduled via (1) cron job and (2) webhook triggers we should ensure we haven't already added the commit. GitHub will retry commits if cocoon sends a non-2** response, so we should handle this case gracefully.

Reply:
That makes sense. I guess we can still guarantee the order of commits with these two scheduling logics, right?